docs/runtime/ir-caching.md

```diff
@@ -0,0 +1,209 @@
+---
+layout: developer-doc
+title: IR Caching in the Enso Compiler
+category: runtime
+tags: [runtime, caching]
+order: 10
+---
+
+# IR Caching in the Enso Compiler
+
+One of the largest pain points for users of Enso at the moment is the fact that
+it has to precompile the entire standard library on every project load. This is,
+in essence, due to the fact that the current parser is abysmally slow, and
+incredibly demanding. The obvious solution to improve this is to take the parser
+out of the equation in its entirety, by serialising the parser's output.
+
+To that end, we want to serialise the Enso IR to a format that can later be read
+back in, bypassing the parser entirely. Furthermore, we can move the boundary at
```

Comment:
I would stress out more that moving the boundary is something planned for later.